db/1.14/modules/ROOT/pages/database-connector-input-output-parameters.adoc

```diff
@@ -0,0 +1,219 @@
+			</db:sql>
+			<db:in-out-parameters >
+				<db:in-out-parameter key="updatedStockLevel" />
+				<db:in-out-parameter key="productId" value="#[payload.productId]" />
+				<db:in-out-parameter key="quantity" value="#[payload.quantity]" />
+			</db:in-out-parameters>
+----
+
+== Configure Custom Types
+
+The ability to use custom types as parameters in stored procedures depends on the database system you are using and the capabilities of the Database Connector or framework you are working with. 
+Exercise caution when you work with an Oracle database vendor and use the *Stored procedure* operation with custom database types. 
+When you define the stored procedure within a package, set the column type of a custom type parameter in the connector connection settings:
+
+. In Studio, open the Database Config Global Element Properties window.
+. Click the *Advanced* tab.
+. Set *Column types* to *Edit inline*.
+. Click the plus sign to open the *Column types* window.
+. In the *Column types* window, set *Id* and *Type name* fields with your desired values.
+. Click *OK*.
+
+image::database-column-type.png[Column type configuration window]
+
+In the *Configuration XML* editor window, the configuration looks like this:
+
+[source,xml,linenums]
+----
+<db:column-types >
+<db:column-type id="2003" typeName="MY_CUSTOM_PKG.MY_CUSTOM_TYPE" />
+</db:column-types>
+----
+
+It remains a good practice to define the *Column types* of a custom type parameter, even if the stored procedure is not part of a package but is directly defined.
+
+== Configuration Limitations
+Consider the following limitations when you configure *In out parameters*.
+
+=== Variable Values Reflection
+
+This is a limitation of the Mule runtime engine (Mule) and applies to all database vendors. You might expect that if you pass a variable with a given set of values to the stored procedure as *In out parameters*, the variable can contain modified values after the operation completes. This is not the case, as these variables are not modified. This is how Mule works by design. The modified values are part of the operation's payload instead. To illustrate this limitation, consider the following scenario:
+
+A parameter registered as *In out parameters* has a value stored in a variable named `foo`. If `vars.foo` is passed to the stored procedure, `vars.foo` retains its original value after the operation is invoked. This behavior can have a challenge, especially when chaining stored procedures. You might expect that when you call the stored procedure A, the value of "vars.foo" would change, allowing them to subsequently call the stored procedure B with the already modified values. 
+However, in reality, the modified data is returned as an entry in the payload after the operation executes. The payload contains modified values which the stored procedure may produce. Consequently, chaining of stored procedures is feasible by invoking only one stored procedure per operation
```

Comment:
```suggestion
However, in reality, the modified data is returned as an entry in the payload after the operation executes. The payload contains modified values that the stored procedure can produce. Consequently, chaining stored procedures is feasible by invoking only one stored procedure per operation.
```

Reply:
Do you need the second sentence? It seems redundant. Maybe you can combine it with the first one if you can't remove it.